internal/pkg/service/stream/plugin/sink.go

```diff
@@ -0,0 +1,26 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/pipeline"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+func (p *Plugins) RegisterSinkPipelineOpener(fn pipeline.Opener) {
+	p.sinkPipelineOpeners = append(p.sinkPipelineOpeners, fn)
+}
+
+func (p *Plugins) OpenSinkPipeline(ctx context.Context, sink definition.Sink) (pipeline.Pipeline, error) {
+	for _, fn := range p.sinkPipelineOpeners {
+		p, err := fn(ctx, sink)
+		if err != nil {
+			return nil, err
+		}
+		if p != nil {
+			return p, nil
+		}
```

Comment:
Go standard says, taht when `err != nil` the `p` is `nil`. When `err == nil` the `p != nil`. Therefore the if is not needed, so it is not according the standard flow check. Why the function can return `p and err both nil`?

Reply:
https://github.com/keboola/keboola-as-code/blob/7f440142904dffdfb412f2cd1e0c316a922014a2/internal/pkg/service/stream/sink/pipeline/pipeline.go#L26-L29

I will copy the comment also here.

Reply:
https://github.com/keboola/keboola-as-code/pull/1903/commits/373b6b71c26f9d085ac7f6cb3f5315a121d04d87

Reply:
https://github.com/keboola/keboola-as-code/pull/1906

Reply:
:heavy_check_mark: